python/deformed_mappings/README.md

```diff
@@ -0,0 +1,38 @@
+# Deformed Mapping Code Generation
+## Introduction
+
+In FEM codes there exist (non)-linear maps $X(\xi)$ from reference element space ($\xi$) to physical space.
+The non-linearity of these maps complicate the process of finding a mesh element which contains a given physical point. 
+This "cell binning" problem is formulated as follows; given an input point $x$, find an element e and reference position $\xi$ such that $X_e(\xi) = x$.
+For each candidate mesh element $e$ a Newton iteration is applied to evaluate $\xi = X_e^{-1}(x)$.
+If $\xi$ falls inside the reference element, for the geometry type of $e$, then $e$ is the element that "contains" the point $x$.
+
+Each (linear sided) element type has a different map $X_e$ which is a function of the vertices of a particular instance of the element.
+The functional form of the map $X$ ranges between linear and tri-linear depending on the element type.
+This repository symbolically represents the maps $X$ for each of the 3D element types and uses the automatic differentiation and code generation capabilities of Sympy to generate the C++ code required to perform the Newton iterations.
+
+## Installation
+The requirements are listed in the `requirements.txt` file and can be installed into a Python virtual environment with
+```
+pip install -r requirements.txt
```

Comment:
It would be better to install the dependencies via spack. Numpy should already be available, so just needs SymPy.

Reply:
I agree - I think this is something to defer until users are running python as part of the standard build though.